### DISCUSSION TITLE
Would be helpfull to know which voices have support for pitch paramter

### DISCUSSION THREAD

**Discussion by Falcosc** [HadrienGardeur/web-speech-recommended-voices#34] [Ideas]
I don't know how many people would need this feature, but for me, pitch is interesting to create multiple different voices.
Seems like Microsoft Online Voices don't have support for pitch. Which is fine, since we have so many of them. Maybe pitch doesn't work there because these voices support emotional pronunciation https://learn.microsoft.com/en-us/azure/ai-services/speech-service/speech-synthesis-markup-voice#use-speaking-styles-and-roles I don't know if this API is already available for the web.

Having Microsoft here really saves me with their free high quality voices.
For the end user, it would be only useful to avoid getting confused with not working settings.
Currently, I just take the obvious shortcut over the name: https://github.com/Falcosc/PSVR2-Twitch/commit/1defaa2936fee7159fe9a2c57c51b524c7555247

**Comment by HadrienGardeur** [comment]
I wasn't aware that some voices didn't support pitch control. I've included `pitch` and `rate` when I first encountered voices that felt slightly off with the default values for pitch/speed rate, but I haven't had time to document and test this as much as I would like.

I'm going to create a new task for adding `pitchControl` as a boolean in the JSON Schema and to document this property on all Microsoft Online voices. Using this new boolean, implementers could disable pitch control accordingly. It's better to disable pitch (or grey it out) than display a component that won't change anything for the user.

It also feels worth filing a bug for this one too.

**Comment by HadrienGardeur** [comment]
> Maybe pitch doesn't work there because these voices support emotional pronunciation https://learn.microsoft.com/en-us/azure/ai-services/speech-service/speech-synthesis-markup-voice#use-speaking-styles-and-roles I don't know if this API is already available for the web.

These styles are per voice and seem to limit the usual SSML controls:

> Multilingual voices don't fully support certain SSML elements, such as break, emphasis, silence, and sub.

These next generation voices are created using ML, this might explain why they behave differently than more traditional TTS voices. It's possible that Microsoft has to train these voices for each style, but cannot really provide fine-tuned controls over them. Eleven Labs seems to have the same limitations as well.

**Comment by HadrienGardeur** [comment]
I'm closing that discussion, since this has already been documented.
